api/proto/teleport/recordingencryption/v1/recording_encryption_service.proto

```diff
@@ -0,0 +1,86 @@
+// Copyright 2025 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.recordingencryption.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/recordingencryption/v1;recordingencryptionv1";
+
+// RecordingEncryption provides methods to manage cluster encryption configuration resources.
+service RecordingEncryptionService {
+  // CreateUpload begins a multipart upload for an encrypted recording. The
+  // returned upload ID should be used while uploading parts.
+  rpc CreateUpload(CreateUploadRequest) returns (CreateUploadResponse);
+  // UploadPart uploads a part to the given upload ID.
+  rpc UploadPart(UploadPartRequest) returns (UploadPartResponse);
+  // CompleteUploadRequest marks a multipart upload as complete.
+  rpc CompleteUpload(CompleteUploadRequest) returns (CompleteUploadResponse);
+}
+
+// An Upload represents a multipart upload for an encrypted session.
+message Upload {
+  // UploadID identifies an upload for a given encrypted session.
+  string upload_id = 1;
+  // SessionID of the associated session.
+  string session_id = 2;
+  // InitiatedAt captures the time that the multipart upload was initiated.
+  google.protobuf.Timestamp initiated_at = 3;
+}
+
+// CreateUploadRequest contains the session ID to be used while initializing the upload.
+message CreateUploadRequest {
+  // SessionID associated with the recording being uploaded.
+  string session_id = 1;
+}
+
+// CreateUploadResponse contains the upload ID to be used when uploading parts.
+message CreateUploadResponse {
+  // Upload represents an encrypted session upload.
+  Upload upload = 1;
+}
+
+// UploadPartRequest is an indivdual part to be uploaded.
+message UploadPartRequest {
+  // Upload represents the encrypted session to upload the part to.
+  Upload upload = 1;
+  // PartNumber is the ordered index applied to the part.
+  int64 part_number = 2;
+  // Part is the encrypted part of session recording data being uploaded.
+  bytes part = 3;
+}
+
+// UploadPartResponse contains the part index that was uploaded with its
+// associated e-tag.
+message UploadPartResponse {
+  // PartIndex is the ordered index applied to the part.
+  int64 part_number = 1;
```

Comment:
Is this allowed to differ from the `part_number` in the request? Was this called part index initially?

Reply:
It was called part index. I switched it to "number" when I realized I diverged from  how we refer to it elsewhere. I don't think it necessarily has to match what was given in the request as long as the final list of parts provided in `CompleteUpload` match what were actually uploaded.